### PR TITLE
[Debugger] Add global variable support

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose,
This PR 
 - introduces debugging support for global/constant variable scopes.
 - updates expression evaluation engine to support global variables.
 - adds integration tests for global variable related contexts.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/25275.

## Remarks

Please note that few of the test scenarios are currently disabled due to the following minor runtime issues which were captured while working the $subject and should be enabled once the relevant fixes are merged to the runtime.
 - https://github.com/ballerina-platform/ballerina-lang/issues/26122 - Ballerina runtime error due to naming conflicts between user defined and generated variable names.
 - https://github.com/ballerina-platform/ballerina-lang/issues/26139 - Inconsistent behavior when generating byte code for ballerina constant definitions.

## Samples
Please refer below scenarios with global variable and constant definitions.

Screenshot from 2020-09-29 16-12-00

Screenshot from 2020-09-29 16-03-01 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java [hunk 1]
```diff
@@ -356,15 +360,18 @@ private void setBreakpoints(List<BallerinaTestDebugPoint> breakPoints) throws Ba
...
+                scopeArgs.setFrameId(-stackFrames[0].getId());
             }
-
-            scopeArgs.setFrameId(stackFrames[0].getId());
             ScopesResponse scopesResp = DebugHitListener.connector.getRequestManager().scopes(scopeArgs);
             variableArgs.setVariablesReference(scopesResp.getScopes()[0].getVariablesReference());
             VariablesResponse variableResp = DebugHitListener.connector.getRequestManager().variables(variableArgs);
-            return variableResp.getVariables();
-
+            Arrays.stream(variableResp.getVariables()).forEach(variable -> variables.put(variable.getName(), variable));
```
Comment (nadeeshaan) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java:373]: As a practice, let's try to get each of the stream operations to a new line

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java [hunk 2]
```diff
@@ -356,15 +360,18 @@ private void setBreakpoints(List<BallerinaTestDebugPoint> breakPoints) throws Ba
                     .stackTrace(stackTraceArgs);
             StackFrame[] stackFrames = stackTraceResp.getStackFrames();
             if (stackFrames.length == 0) {
-                return new Variable[0];
+                return variables;
+            }
+            if (scope == VariableScope.LOCAL) {
```
Comment (nadeeshaan) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java:365]: May be we can use the conditional operator here

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java]: Kept it inline as there's only one stream operation in here. changed with db9e03c.

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java (hunk 2 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java]: Done via db9e03c

**Review by praveennadarajah — APPROVED** [review]

**Pull request merged**